packages/helpermodules/simcount.py

```diff
@@ -0,0 +1,180 @@
+                watt1 = watt1 + stepsize
+                if stepsize < 0:
+                    watt1 = max(watt1, present_power_all)
+                else:
+                    watt1 = min(watt1, present_power_all)
+                seconds1 = seconds1 + 1
+            rest = deltasec - deltasectrun
+            seconds1 = seconds1 - 1 + rest
+            if rest > 0:
+                watt1 = int(watt1 * rest)
+                if watt1 < 0:
+                    wattnegh = wattnegh + watt1
+                else:
+                    wattposh = wattposh + watt1
+            wattposkh = wattposh/3600
+            wattnegkh = (wattnegh*-1)/3600
+            if ramdisk == True:
+                f = open('/var/www/html/openWB/ramdisk/'+pref+'watt0pos', 'w')
+                f.write(str(wattposh))
+                f.close()
+                if wattposh != importtemp:
+                    pub.pub_single("openWB/pv/WHImported_temp", wattposh, no_json=True)
+                f = open('/var/www/html/openWB/ramdisk/'+pref+'watt0neg', 'w')
+                f.write(str(wattnegh))
+                f.close()
+                if wattnegh != exporttemp:
+                    pub.pub_single("openWB/pv/WHExport_temp", wattnegh, no_json=True)
+            else:
+                pub.pub(topic+"module/simulation/present_imported", wattposh)
+                pub.pub(topic+"module/simulation/present_exported", wattnegh)
+            return wattposkh, wattnegkh
+        else:
+            value1 = "%22.6f" % sim_timestamp
+            if ramdisk == True:
+                f = open('/var/www/html/openWB/ramdisk/'+pref+'sec0', 'w')
+                f.write(str(value1))
+                f.close()
+                f = open('/var/www/html/openWB/ramdisk/'+pref+'wh0', 'w')
+                f.write(str(watt2))
+                f.close()
+            else:
+                pub.pub(topic+"module/simulation/sim_timestamp", value1)
+                pub.pub(topic+"module/simulation/present_power_all", present_power_all)
+            return 0, 0
+    except Exception as e:
+        log.log_exception_comp(e, ramdisk)
+
+def restore(value):
+    signal.signal(signal.SIGALRM, abort)
+    signal.alarm(3)
+    try:
+        if value == "watt0pos":
+            temp = subscribe.simple("openWB/pv/WHImported_temp", hostname="localhost")
+        else:
+            temp = subscribe.simple("openWB/pv/WHExport_temp", hostname="localhost")
+    except:
+        pass
+    ra='^-?[0-9]+$'
+    if re.search(ra, temp) == None:
+        temp = "0"
+    f = open('/var/www/html/openWB/ramdisk/'+pref+value, 'w')
+    f.write(str(temp))
+    f.close()
+    if value == "watt0pos":
+        log.log_1_9("loadvars read openWB/pv/WHImported_temp from mosquito "+str(temp))
+    else:
+        log.log_1_9("loadvars read openWB/pv/WHExport_temp from mosquito "+str(temp))
+    return temp
+
+def abort(signal, frame):
+    raise Exception
+
+if __name__ == "__main__":
+    try:
+        watt2 = int(sys.argv[1])
```

Comment:
Hier wird eine globale Variable angelegt. Auf diese wird auch tatsächlich in Zeile 140 zugegriffen. Das macht die Zusammenhänge eher undurchsichtig. Ich würde gerne anregen die Variable zu inlinen oder das ganze in eine Funktion packen, damit es eine lokale Variable wird.